File: 0974. Subarray Sums Divisible by K/Solution0974.py
from typing import List


class Solution0974:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        m = [0 for i in range(k)]
        m[0] = 1
        count, tot_sum = 0, 0
        for num in nums:
            tot_sum += num
            cur_rem = tot_sum % k
            if cur_rem < 0:
                cur_rem = cur_rem + k
            count += m[cur_rem]
            m[cur_rem] += 1
            
        return count

File: 0974. Subarray Sums Divisible by K/Solution0974.java
public class Solution0974 {
    public int subarraysDivByK(int[] nums, int k) {
		//stores the remainders
        int m[]=new int[k];
        int sum=0;
        int count=0;
        m[0]=1;
        for(int x: nums){
            sum+=x;
            int rem=sum%k;
            if(rem<0) rem+=k;
            count+=m[rem];
            m[rem]++;
        }
        return count;
    }
}